common/datavalues/src/types/data_type_coercion.rs

```diff
@@ -194,6 +194,9 @@ pub fn numerical_arithmetic_coercion(
             construct_numeric_type(true, has_float, next_size(max_size))
         }
         DataValueArithmeticOperator::Div => Ok(DataType::Float64),
+        DataValueArithmeticOperator::IntDiv => {
+            construct_numeric_type(has_signed, has_float, max_size)
```

Comment:
How about this:

We set has_float to be false even has_float is true here, so the result will be integer all the way.
Then we can remove the match case above.

Reply:
In this case we will get incorrect result for, e.g. SELECT 7 DIV 2.99' - it is going to be 3 instead of 2 because 2.99 will cast to 2 before DIV